packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart

```diff
@@ -432,6 +451,62 @@ class FlutterDebugAdapter extends FlutterBaseDebugAdapter {
     }
   }
 
+  /// Handles incoming reverse requests from `flutter run --machine`.
+  ///
+  /// These requests are usually just forwarded to the client via an event
+  /// (`flutter.forwardedRequest`) and responses are provided by the client in a
+  /// custom event (`flutter.forwardedRequestResponse`).
+  void _handleJsonRequest(
+    Object id,
+    String method,
+    Map<String, Object?>? params,
+  ) {
+    /// A helper to send a client response to Flutter.
+    void sendResponseToFlutter(Object? id, Object? value, { bool error = false }) {
+      sendFlutterMessage(<String, Object?>{
+        'id': id,
+        if (error)
+          'error': value
+        else
+          'result': value
+      });
+    }
+
+    // Set up a completer to forward the response back to `flutter` when it arrives.
+    final Completer<Object?> completer = Completer<Object?>();
+    _reverseRequestCompleters[id] = completer;
+    completer.future
+        .then((Object? value) => sendResponseToFlutter(id, value))
+        .catchError((Object? e) => sendResponseToFlutter(id, e.toString(), error: true));
+
+    if (_requestsToForwardToClient.contains(method)) {
+      // Forward the request to the client in an event.
+      sendEvent(
+        RawEventBody(<String, Object?>{
+          'id': id,
+          'method': method,
+          'params': params,
+        }),
+        eventType: 'flutter.forwardedRequest',
+      );
+    } else {
+      completer.completeError('Unknown request method "$method".');
```

Comment:
I think this is eventually going to be an analysis error for `only_throw_errors` (right now the lint is only enforced for errors thrown with the `throw` keyword). Any reason not to wrap this in an `Exception`?

Reply:
Done!